src/main.rs

```diff
@@ -1036,6 +1043,9 @@ fn drawe(app: &mut App, gfx: &mut Graphics, plugins: &mut Plugins, state: &mut O
         if let Some(id) = state.filebrowser_id.take() {
             ctx.memory_mut(|w| w.open_popup(Id::new(&id)));
         }
+        
+        // Show position dialog if it's open
```

Comment:
I don't like that the function handles the visibility. You should do what your comment says and check if the dialog is open. I also don't like that application state is used for visibility. You should use egui memory for UI element visibility (ctx.memory for example).